packages/react-i18n/src/index.tsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { __, _n, _nx, _x } from '@wordpress/i18n';
+
+export const I18nContext = React.createContext< undefined | string >( undefined );
```

Comment:
Can we make the default value be `'en'`? Even if you use `undefined`, it effectively means `en`, because the strings in the source are left unchanged, i.e., in English.

Reply:
🤔 I'm not sure. Even Calypso has a configurable default locale:

https://github.com/Automattic/wp-calypso/blob/5a5fb58300b6b19848992517fffc36fc4e4b4fb9/config/_shared.json#L23

Reply:
Yeah, but I think that config value is not very useful and I'd like to eventually remove it.

Think about it this way: the default language is the one that the strings in the source code are written it:
```js
__( 'Stats and Insights' )
```
These strings will be displayed in the UI when the `__` function is identity. And they are always in _some_ language, it's never anything that has `undefined` or `null` meaning.

The default language is the language that the sources are written it. It can theoretically be non-`en`, but I don't think that any part of our i18n toolchain supports that.

Reply:
Do folks using wp-i18n ever use string keys? E.g. `__( 'stats.heading' )` gets translated to `'Stats and Insights'` when locale is English.

Reply:
@simison Last I recall, there's no localization data assigned for the default locale, so it will always use whatever's passed as the string argument in those cases (tl;dr: No).

Reply:
> the default language is the one that the strings in the source code are written it

Sure, that makes sense. But it's convention that it would be English, I could write my source in any language and provide English via translations if I desired. `undefined` seems like a good way of saying "do not translate." Another option might be a fixed string like `"default"`.

Reply:
> But it's convention that it would be English, I could write my source in any language and provide English via translations if I desired.

Then, if you write your source in Spanish, for example, then the locale in the context is supposed to be `es`, not `undefined`. The UI, if not translated, is in Spanish. The value of the locale can be used also by components like `<MomentProvider>` to format dates in Spanish. `undefined` is not a locale name that's helpful for moment.js.

`undefined` is meaningful in the case where message IDs are used, as @simison mentioned. `__( 'stats.heading' )` doesn't produce anything usable until a translation file is provided. Message IDs are common on some platforms (Win32 or Java comes to mind), but WordPress doesn't use them.

I'd advise strongly to use `en` as a default, because that's in fact the WordPress default.